### PR TITLE
Refactoring validation of items and time_steps

### PR DESCRIPTION
**Pull request by jsmariegaard**
Refactoring the way we validate requested items and timesteps. no api change. no new functionality.

### CODE REVIEW

**Comment by jsmariegaard** [comment]
I have worked a bit to improve the test coverage too (even though it was not really the purpose of this PR)

File: mikeio/dfs.py [hunk 1]
```diff
@@ -58,15 +59,15 @@ def read(self, items=None, time_steps=None):
...
 
-        items, item_numbers, time_steps = self._get_valid_items_and_timesteps(
-            items, time_steps
-        )
+        item_numbers = valid_item_numbers(self._dfs.ItemInfo, items)
+        n_items = len(item_numbers)
+
+        time_steps = valid_timesteps(self._dfs.FileInfo, time_steps)
 
         for t in time_steps:
```
Comment (ecomodeller) [inline mikeio/dfs.py:67]: Is this still necessary?

File: mikeio/dfsutil.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    item_lookup = {name: i for i, name in enumerate(names)}
+    try:
+        item_numbers = [item_lookup[x] for x in item_names]
+    except KeyError:
+        raise KeyError(f"Selected item name not found. Valid names are {names}")
+
+    return item_numbers
+
+
+def get_item_info(dfsItemInfo, item_numbers):
```
Comment (ecomodeller) [inline mikeio/dfsutil.py:109]: Add default argument to item_numbers (default should be all items)

File: mikeio/dfs.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 )
 from .eum import ItemInfo, TimeStepUnit, EUMType, EUMUnit
 from .custom_exceptions import DataDimensionMismatch, ItemNumbersError
+from .dfsutil import valid_item_numbers, valid_timesteps, get_item_info
```
Comment (ecomodeller) [inline mikeio/dfs.py:16]: Rename dfsutil to _dfsutil to indicate it is for internal use?

**Review by ecomodeller — APPROVED** [review]

File: mikeio/dfs.py (hunk 1 above)
Comment (jsmariegaard) [reply mikeio/dfs.py:67]: I assume that you're referring to the lines below:

        for t in time_steps:
            if t > (self.n_timesteps - 1):
                raise IndexError(f"Timestep {t} is > {self.n_timesteps-1}")

which I have now deleted

File: mikeio/dfsutil.py (hunk 2 above)
Comment (jsmariegaard) [reply mikeio/dfsutil.py]: Done

File: mikeio/dfs.py (hunk 3 above)
Comment (jsmariegaard) [reply mikeio/dfs.py:16]: Maybe. But perhaps **better to rename the methods to underscore** _variants()? I looked at this https://stackoverflow.com/questions/3602110/python-private-module-in-a-package and https://stackoverflow.com/questions/55157525/leading-underscore-before-the-name-of-python-module  "In the Python standard library, a leading underscore in a module name may denote an accelerator module: a module written in C that provides a more performant implementation of the pure Python version" https://www.python.org/dev/peps/pep-0399/#details

Comment (jsmariegaard) [reply mikeio/dfs.py]: agreed to rename methods instead of module (on phone)

**Pull request merged**